acceptance_tests/features/environment.py

```diff
@@ -92,8 +92,8 @@ def after_scenario(context, scenario):
 
     leftover_messages = purge_outbound_topics()
     if leftover_messages:
-        test_helper.fail(f'There are left over messages on the following subscriptions: {leftover_messages}, see logs '
-                         f'above for details.')
+        logger.warn(f'There are left over messages on the following subscriptions: {leftover_messages}, see logs '
+                    f'above for details.')
```

Comment:
I don't think this will address the failures on the linked ticket, this is the post-scenario check to ensure nothing is left by the scenario, where the failure linked on the ticket is coming from the before scenario hook on the first test.

I think we would actually want to keep this particular failure, as it prevents us from writing scenarios which leave un consumed messages

Reply:
The `_ack_all_on_subscription` in `before_scenario`  currently just logs out the error for each message and carries on, which was why this method in `after_scenario` was changed to prevent failure. 

Following discussion, we're now going to retry purging the pub/sub at the beginning in `before_all` instead